Yoast/wordpress-seo
# Create indexing helper to get and set options

## Context


* Separation of concerns: integrations that have a need to set or get options, shouldn't know about the inner workings of this. They can now use the indexing helper (and the existing indexable helper) to get and set options.

## Summary


This PR can be summarized in the following changelog entry:

* Introduces an indexing helper which holds methods to get and set options in the database regarding the indexing process.

## Relevant technical choices:

* I also moved the setting/getting of two options (`indexing_first_time ` and `indexables_indexation_completed`) that are not mentioned in the issue to helpers, for the same reasons as the two options that are mentioned in the issue.
* The `indexables_indexation_completed` option is set in the `indexable-helper` rather than the `indexing-helper` (because it only concerns a part of the indexing process, related to the indexables).
* I removed the setting of `indexation_warning_hide_until` from `wp-seo-main`, because we recently removed all the "Remind me in 7 days" functionality. This line was probably left in the code because [it was merged](https://github.com/Yoast/wordpress-seo/commit/5794cea83f866f3d76f639c04f0a134e8e1799ad) around the same time the functionality was removed in a different PR (https://github.com/Yoast/wordpress-seo/pull/16204).
* Also in `wp-seo-main`, I decided not to touch `WPSEO_Options::set( 'indexing_reason', 'first_install' );` because this code is executed so early (on plugin activation). I wasn't sure whether the helpers would already be available.
* I removed the `indexing-complete-test`, which was a duplicate of `indexing-complete-action-test`.
* The automatic CS fixer has also touched some files that are otherwise unrelated to this PR.

## Test instructions

This PR can be tested by following these steps:

* Reset the indexables using the Test Helper.
* In your database, in `wp_options`, `wpseo`, check that these options have the following values:
    * `indexing_first_time` --> `1`.
    * `indexation_started` --> `0`.
    * `indexing_reason` --> `""`.
    * `indexables_indexation_completed` --> `0`.
* Go to the Tools page.
* In the console, type `window.yoastIndexingData` and press return.
* Inspect the `yoastIndexingData` object and see that `firstTime` is set to `1`.
* Run the indexing process and stop it halfway through.
* Check that the `indexation_started` option has as its value a large number, indicating a point in time.
* Let the indexing process run to the end. See that the options are now:
    * `indexing_first_time` --> `0`.
    * `indexation_started` --> `0`.
    * `indexing_reason` --> `""`.
    * `indexables_indexation_completed` --> `1`.
* See that the notification is gone.
* Change your permalink settings.
* See that the notification is back, and holds the correct text ("Because of a change in your permalink structure, some of your SEO data needs to be reprocessed.").
* Reset the indexables.
* See the indexing notification is present. 
    * If not, deactivate and activate the plugin. This is a workaround to get the notification to show again, because at the moment there is a [bug in the Test Helper](https://yoast.atlassian.net/browse/P2-392) (the indexing reason is not set upon resetting the indexables, which causes the notification not to show).
* Hide the indexing notification.
* Refresh the Yoast dashboard, see that the indexing notification is still hidden. 
    * If not, the code from https://github.com/Yoast/wordpress-seo/pull/16226 still needs to be merged into this branch before you can continue testing.
* Introduce an error in de indexing process, for example by adding the following line to the `index` function in `indexable-post-indexation-action.php`: `throw new \Exception( 'An error occurred in the indexing process.' );`.
* Run the indexing process, see that it fails. 
* Check that the `indexing_reason` option has as its value `indexing_failed`.
* Check that the indexing notification is present again.

**Explanation**
The above test instructions both apply to this branch and to `trunk`. For a successful test, the behavior in this branch should not have changed as compared to `trunk`.  This is because it is not a bugfix, but a code refactor.

## UI changes

## Documentation

## Quality assurance


Fixes https://yoast.atlassian.net/browse/P2-433
